pkgs/os-specific/linux/service-wrapper/default.nix

```diff
@@ -0,0 +1,280 @@
+{ stdenv, fetchurl, systemd, coreutils,
+  src ? null }:
+
+stdenv.mkDerivation rec {
+  name = "service-wrapper-${version}";
+  version = "16.04.0"; # Ajar to Ubuntu Release
+
+  inherit src;
+
+  unpackCmd = ''
```

Comment:
unpackCmd, dontBuild, installPhase and postFixup can be replaced by single phase - `buildCommand`

But in that case you probably would find it better to use `lib.runCommand` or `lib.writeScriptBin`